### PR TITLE
lease: delete items when the lease is revoked.

### PR DESCRIPTION
**Pull request by xiang90**
Add minimum KV interface and implmement the deleting mechanism for
revoking lease.

### CODE REVIEW

File: lease/lessor.go [hunk 1]
```diff
@@ -85,7 +100,10 @@ func (le *lessor) Revoke(id uint64) error {
 
 	delete(le.leaseMap, l.id)
 
-	// TODO: remove attached items
+	for item := range l.itemSet {
+		le.KV.DeleteRange([]byte(item.key), []byte(item.endRange))
```
Comment (yichengq) [inline lease/lessor.go]: I don't understand this. The range [key, endRange] may cover the keys that don't want to be removed:
1. Another users adds a new key in this range when the lease is going on
2. the key is used in another lease

Maybe I don't understand lease mechanism yet. Could you explain more?

Comment (xiang90) [inline lease/lessor.go]: you can lease on a range or on a key.  When a lease expires, the lessor will remove the leased items from the KV.

I think this is good enough. And your concern will complicate the issue a lot. We will not do per-key ownership.

Comment (xiang90) [inline lease/lessor.go]: You can even lease on non existing key and range. This is by design and solves a lot of real problems in practice. If you cannot lease on non-existing key, there will be a race between creating key and creating lease. If one dies before creating the lease, then the key will never be removed.

Comment (yichengq) [inline lease/lessor.go]: ok. good to know.

File: lease/lessor.go [hunk 2]
```diff
@@ -26,6 +26,15 @@ var (
 	minLeaseTerm = 5 * time.Second
 )
 
+// DeleteableKV defines a KV interface with DeleteRange method.
+// We define this interface only for lessor to limit the number
+// of methods of storage.KV to what lessor actually needs.
+//
+// Having a minimum interface makes testing easy.
+type DeleteableKV interface {
```
Comment (yichengq) [inline lease/lessor.go]: The name is not good IMO. For DeleteableKV, i would expect it to be a full KV with deletion functionality. If it is just for fast move, i am ok with that.

Comment (xiang90) [inline lease/lessor.go]: What is a good name in you opinion? Please give me some suggestion.

Comment (yichengq) [inline lease/lessor.go]: i cannot find a good one now. maybe simplify it to KV.

Comment (jonboulle) [inline lease/lessor.go]: how about just DeleteableRange?

Comment (xiang90) [inline lease/lessor.go]: @jonboulle I am fine with DeleteableRange

Comment (yichengq) [inline lease/lessor.go]: me too.

**Comment by xiang90** [comment]
@jonboulle @yichengq PTAL.

**Comment by yichengq** [comment]
lgtm

**Pull request merged**